server/src/test/java/org/elasticsearch/index/mapper/DynamicMappingTests.java

```diff
@@ -769,12 +769,21 @@ public void testDynamicTemplateOrder() throws IOException {
         assertThat(index.mapperService().fullName("foo"), instanceOf(KeywordFieldMapper.KeywordFieldType.class));
     }
 
-    public void testMappingVersionAfterDynamicMappingUpdate() {
+    public void testMappingVersionAfterDynamicMappingUpdate() throws Exception {
         createIndex("test", client().admin().indices().prepareCreate("test").addMapping("type"));
         final ClusterService clusterService = getInstanceFromNode(ClusterService.class);
         final long previousVersion = clusterService.state().metaData().index("test").getMappingVersion();
         client().prepareIndex("test", "type", "1").setSource("field", "text").get();
-        assertThat(clusterService.state().metaData().index("test").getMappingVersion(), equalTo(1 + previousVersion));
+        // This assertBusy is necessary. When a mapping update is needed as a document is indexed,
```

Comment:
Thanks for the detailed message, TIL.  However, I think the commit message is sufficient for the detailed explanation. Perhaps a shorter version like:
`//ensure that cluster state has been updated`

Also, nitpick:  `s/mapping conflict/required mapping update`